obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+        mag_2 = ''
+        mag_2_type = ''
+        mag_2_agency = ''
+    try:
+        mag_3 = '{0:.1f}'.format(event.magnitudes[2].mag) or ''
+        mag_3_type = _evmagtonor(event.magnitudes[2].magnitude_type) or ''
+        if event.magnitudes[2].creation_info:
+            mag_3_agency = event.magnitudes[2].creation_info.agency_id or ''
+        else:
+            mag_3_agency = ''
+    except IndexError:
+        mag_3 = ''
+        mag_3_type = ''
+        mag_3_agency = ''
+    # Work out how many stations were used
+    if len(event.picks) > 0:
+        stations = [pick.waveform_id.station_code for pick in event.picks]
+        ksta = str(len(list(set(stations))))
+    else:
+        ksta = ''
+    sfile.write(' ' + str(evtime.year) + ' ' +
+                str(evtime.month).rjust(2) +
+                str(evtime.day).rjust(2) + ' ' +
+                str(evtime.hour).rjust(2) +
+                str(evtime.minute).rjust(2) + ' ' +
+                str(evtime.second).rjust(2) + '.' +
+                str(evtime.microsecond).ljust(1)[0:1] + ' ' +
+                evtype.ljust(2) + lat.rjust(7) + ' ' + lon.rjust(7) +
+                depth.rjust(5) + agency.rjust(5) + ksta.rjust(3) +
+                timerms.rjust(4) +
+                mag_1.rjust(4) + mag_1_type.rjust(1) +
+                mag_1_agency[0:3].rjust(3) +
+                mag_2.rjust(4) + mag_2_type.rjust(1) +
+                mag_2_agency[0:3].rjust(3) +
+                mag_3.rjust(4) + mag_3_type.rjust(1) +
+                mag_3_agency[0:3].rjust(3) + '1' + '\n')
+    # Write line 2 of s-file
+    sfile.write(' ACTION:ARG ' + str(datetime.datetime.now().year)[2:4] + '-' +
+                str(datetime.datetime.now().month).zfill(2) + '-' +
+                str(datetime.datetime.now().day).zfill(2) + ' ' +
+                str(datetime.datetime.now().hour).zfill(2) + ':' +
+                str(datetime.datetime.now().minute).zfill(2) + ' OP:' +
+                userid.ljust(4) + ' STATUS:'+'ID:'.rjust(18) +
+                str(evtime.year) +
+                str(evtime.month).zfill(2) +
+                str(evtime.day).zfill(2) +
+                str(evtime.hour).zfill(2) +
+                str(evtime.minute).zfill(2) +
+                str(evtime.second).zfill(2) +
+                'I'.rjust(6) + '\n')
+    # Write line 3 of s-file
+    for wavefile in wavefiles:
+        sfile.write(' ' + wavefile + '6'.rjust(79-len(wavefile)) + '\n')
+    # Write final line of s-file
+    sfile.write(' STAT SP IPHASW D HRMM SECON CODA AMPLIT PERI AZIMU' +
+                ' VELO AIN AR TRES W  DIS CAZ7\n')
+    sfile.close()
+    # Now call the populatesfile function
+    if len(event.picks) > 0:
+        populatesfile(outdir + '/' + sfilename, event)
+    return str(sfilename)
+
+
+def populatesfile(sfile, event):
+    """
+    Populate a blank nordic format S-file with pick information.
+    arguments required are the filename of the blank s-file and the picks \
```

Comment:
This paragraph should probably go in the description of the `picks` parameter below. It doesn't seem to be about the function in general.